app/controllers/users/phone_confirmation_controller.rb

```diff
@@ -40,11 +40,13 @@ def assign_phone
                           phone_confirmed_at: Time.current)
     end
 
-    def after_confirmation_path
+    def after_phone_confirmation
       if @updating_existing_number
-        profile_path
+        create_user_event(:phone_changed)
+        redirect_to profile_path
       else
-        after_sign_in_path_for(current_user)
+        create_user_event(:phone_confirmed)
+        redirect_to after_sign_in_path_for(current_user)
```

Comment:
I actually had separate PR ready that refactored this stuff but I wasn't able to rebase it easily since Peter's recent refactor.  I'll take another stab it it now.